packages/node/src/indexer/StoreOperations.ts

```diff
@@ -0,0 +1,141 @@
+      hashType: 'sha256',
+    });
+  }
+
+  private operationEntityToBuffer(operation: OperationEntity): Buffer {
+    //expect output: ["Set","StarterEntity", "0xasdasd..."]
+    let operationBuffer: Uint8Array = new Uint8Array();
+    //expect output: blake2AsHex(["id","1", "field1","12300000",...]) = "0xasdasd..."
+    let dataBuffer: Uint8Array = new Uint8Array();
+    operationBuffer = u8aConcat(
+      operationBuffer,
+      Buffer.from(operation.operation),
+    );
+    operationBuffer = u8aConcat(
+      operationBuffer,
+      Buffer.from(operation.entityType),
+    );
+
+    if (operation.operation === OperationType.Remove) {
+      //remove case
+      dataBuffer = u8aConcat(dataBuffer, Buffer.from('id'));
+      dataBuffer = u8aConcat(dataBuffer, Buffer.from('Remove'));
+    } else {
+      const operationModel = this.models.find(
+        ({ name }) => name === operation.entityType,
+      );
+      for (const field of operationModel.fields) {
+        const fieldValue = operation.data[field.name];
+        if (!fieldValue) {
+          continue;
+        } else {
+          switch (field.type) {
+            case 'Date':
+              dataBuffer = u8aConcat(dataBuffer, Buffer.from(field.name));
+              dataBuffer = u8aConcat(
+                dataBuffer,
+                Buffer.from(numberToU8a(fieldValue.getTime())),
+              );
+              break;
+            case 'BigInt':
+              dataBuffer = u8aConcat(dataBuffer, Buffer.from(field.name));
+              dataBuffer = u8aConcat(
+                dataBuffer,
+                Buffer.from(fieldValue.toString()),
+              );
+              break;
+            case 'ID':
+              dataBuffer = u8aConcat(dataBuffer, Buffer.from(field.name));
+              dataBuffer = u8aConcat(
+                dataBuffer,
+                Buffer.from(fieldValue.toString()),
+              );
+              break;
+            case 'Int':
+              dataBuffer = u8aConcat(dataBuffer, Buffer.from(field.name));
+              dataBuffer = u8aConcat(
+                dataBuffer,
+                Buffer.from(numberToU8a(fieldValue.toString())),
+              );
+              break;
+            case 'Boolean':
+              dataBuffer = u8aConcat(dataBuffer, Buffer.from(field.name));
+              dataBuffer = u8aConcat(
+                dataBuffer,
+                Buffer.from(numberToU8a(fieldValue ? 1 : 0)),
+              );
+              break;
+            case 'String':
+              dataBuffer = u8aConcat(dataBuffer, Buffer.from(field.name));
+              dataBuffer = u8aConcat(
+                dataBuffer,
+                Buffer.from(Buffer.from(fieldValue)),
+              );
+              break;
+            default:
+              //JSON
+              dataBuffer = u8aConcat(dataBuffer, Buffer.from(field.name));
+              dataBuffer = u8aConcat(
+                dataBuffer,
+                Buffer.from(JSON.stringify(fieldValue)),
```

Comment:
probably we should check how sequalize handle js object, and we align our code with sequlize's